doc/api/cli.md

```diff
@@ -919,21 +919,27 @@ added: REPLACEME
 
 > Stability: 1.0 - Early development
 
-Use this flag to specify a configuration file that will be loaded and parsed
-before the application starts.
+If the `--experimental-config-file` flag is present, Node.js will look for a
+`node.config.json` file in the current working directory and load it as a
+as configuration file.
+
+If [`--experimental-config-file-path`][] is present, Node.js will look for a
+configuration file at the specified path.
```

Comment:
`--experimental-config-file` should take an argument, like `--experimental-config-file=path-to-file`. This would be consistent with `--env-file=path-to-env-file`. Then when the feature moves out of experimental status it would remain as `--config-file=path-to-file` just like `--env-file`.

I don't think we need a second flag to enable this “automatically load the config file” feature. That aspect can just come later when we unflag this.

Reply:
There is already `--experimental-config-file-path=path` that does exactly that.

Reply:
> There is already `--experimental-config-file-path=path` that does exactly that.

I understand. I feel strongly that `--experimental-config-file-path` should not exist. There should just be one flag, `--experimental-config-path=path-to-file` and that's it.

Reply:
I disagree. When --experimental-config-file is unflagged, there will be --experimental-config-file-path to specify which file to load.
Making a flag that requires a string argument work like a boolean is weird. Not to mention that there should be a --no-experimental-config-file which is obviously boolean.

Reply:
It feels very wrong to me that we would have `--env-file` that takes an argument but `--config-file` that's a boolean.

When this feature is unflagged, `--config-file` should specify what file to load. I'm not sure we really need a way to disable the loading of the default config file, but we could have `--disable-config-file` for that; we already have a few `--disable-*` flags.

Reply:
If we keep only `--experimental-config-file` then it will load the `node.config.json` by default which is not good since imho experimental flags shouldnt do anything implicitly. When we decide to unflag we can rename `experimental-config-file-path` to `experimental-config-file`.

Reply:
First, let's leave `--experimental-config-file=<config>` alone. That's what already landed on `main` and it's good.

As for the automatic loading, what I suggested above was to simply not land that part until the feature is unflagged. Since the feature requires a flag at the moment, there's not much point in automatic loading.

If you really want to land that part now, I'm not opposed, but I think `--experimental-config-file` should stay as it is. How about we create `--enable-default-config-file` to turn on the automatic loading. Then once it's enabled by default, we make `--enable-default-config-file` a no-op and add `--disable-default-config-file`.

Reply:
Sounds good